persistence/src/main/java/edu/softserve/zoo/persistence/provider/impl/HibernatePersistenceProvider.java

```diff
@@ -133,15 +133,15 @@ public boolean delete(Long id, Class type) {
      */
     @Override
     @Transactional(propagation = Propagation.MANDATORY, readOnly = true)
-    public List<T> find(Specification<T> specification) {
+    public <K> List<K> find(Specification<T> specification) {
```

Comment:
so....if I want to use previous functionality, I will get raw List instead of List<T>. 
I expect when I define T - it means I want to received List<T> - not raw List. 

You are breaking generics in this way